extensions/markdown-language-features/src/features/previewManager.ts

```diff
@@ -197,6 +208,30 @@ export class MarkdownPreviewManager extends Disposable implements vscode.Webview
 		return this.registerDynamicPreview(preview);
 	}
 
+	public toggleStaticPreview(uri: vscode.Uri | undefined) {
+		const preview = this._activePreview;
+		if (preview instanceof StaticMarkdownPreview) {
+			vscode.commands.executeCommand('markdown.showTextEditor');
+		} else {
+			vscode.commands.executeCommand('markdown.showStaticPreview', uri);
+		}
+	}
+
+	public openStaticPreview(
+		uri: vscode.Uri,
+	): void {
+		vscode.commands.executeCommand('workbench.action.closeActiveEditor');
+		vscode.commands.executeCommand('vscode.openWith', uri, this.customEditorViewType);
+	}
+
+	public openTextEditor(): void {
+		const preview = this._activePreview;
+		if (preview) {
```

Comment:
Is there a way to check here for the active editor's viewtype? Ideally, would like to only run this when the editor is not a text editor, since doing so just closes the editor.

Reply:
Only the markdown extension knows about the custom editors it creates. There's no general vscode api for accessing them

Checking the `_activePreview` seems correct